TorchSharp/TypeGeneration.tt

```diff
@@ -1115,12 +1174,116 @@ foreach (var type in new [] {
+                new { Name="LeValue", CName= "leValue", ByteResult=true, Doc="Compare a tensor and a scalar for '<=' and produce a byte tensor with the logical result." },
+                new { Name="GtValue", CName= "gtValue", ByteResult=true, Doc="Compare a tensor and a scalar for '>' and produce a byte tensor with the logical result." },
+                new { Name="GeValue", CName= "geValue", ByteResult=true, Doc="Compare a tensor and a scalar for '>=' and produce a byte tensor with the logical result." },
+                new { Name="EqValue", CName= "eqValue", ByteResult=true, Doc="Compare a tensor and a scalar for equality and produce a byte tensor with the logical result." },
+                new { Name="NeValue", CName= "neValue", ByteResult=true, Doc="Compare a tensor and a scalar and inequality and produce a byte tensor with the logical result." },
+                new { Name="LtValueT", CName= "ltValueT", ByteResult=false, Doc="Compare a tensor and a scalar for '<' and produce a tensor with the logical result." },
+                new { Name="LeValueT", CName= "leValueT", ByteResult=false, Doc="Compare a tensor and a scalar for '<=' and produce a tensor with the logical result." },
+                new { Name="GtValueT", CName= "gtValueT", ByteResult=false, Doc="Compare a tensor and a scalar for '>' and produce a tensor with the logical result." },
+                new { Name="GeValueT", CName= "geValueT", ByteResult=false, Doc="Compare a tensor and a scalar for '>=' and produce a tensor with the logical result." },
+                new { Name="EqValueT", CName= "eqValueT", ByteResult=false, Doc="Compare a tensor and a scalar for equality and produce a tensor with the logical result." },
+                new { Name="NeValueT", CName= "neValueT", ByteResult=false, Doc="Compare a tensor and a scalar and inequality and produce a tensor with the logical result." },
+                }){
+                var op = opdoc.Name;
+                var doc = opdoc.Doc;
+                var cname = (string.IsNullOrEmpty(opdoc.CName)) ? op.ToLower() : opdoc.CName;
+                var resultType = opdoc.ByteResult ? "ByteTensor" : tname +"Tensor";
+                var resultHType = opdoc.ByteResult ? "ByteTensor.HType" : "HType";
+            
+#>                
+        [DllImport ("caffe2")]
+        extern static void TH<#=tname#>Tensor_<#=cname#> (<#=resultHType#> result, HType t, <#=ttype#> value);
+        
+        /// <summary>
+        ///   Performs an <#=op#> of the tensor with the provided 
+        ///   <see paramref="src"/> tensor and returns a new tensor with the result.
+        /// </summary>
+        /// <returns>
+        ///   This returns a new tensor with the same shape as the tensor this operates on.
+        /// </returns>
+        public <#=resultType#> <#=op#> (<#=ttype#> src)
```

Comment:
I think we should also surface result as input parameter. This because users may want to do inplace operations (they are supported right?). This comment applies for all the methods having src and target